src/synopsys-action/tools-parameter.ts

```diff
@@ -294,41 +296,40 @@ export class SynopsysToolsParameter {
       }
     }
 
-    // Check and put environment variable for fix pull request
-    if (parseToBoolean(inputs.BLACKDUCK_FIXPR_ENABLED)) {
-      info('Black Duck Fix PR is enabled')
-      blackduckData.data.blackduck.fixpr = this.setBlackDuckFixPrInputs()
-      blackduckData.data.github = this.getGithubRepoInfo()
-    } else {
-      // Disable fix pull request for adapters
-      blackduckData.data.blackduck.fixpr = {enabled: false}
-    }
-
-    if (parseToBoolean(inputs.BLACKDUCK_PRCOMMENT_ENABLED)) {
-      info('Black Duck PR comment is enabled')
-      blackduckData.data.github = this.getGithubRepoInfo()
-      blackduckData.data.blackduck.automation.prcomment = true
+    if (isPullRequestEvent()) {
+      /** Set Black Duck PR comment inputs in case of PR context */
+      if (parseToBoolean(inputs.BLACKDUCK_PRCOMMENT_ENABLED)) {
+        info('Black Duck PR comment is enabled')
+        blackduckData.data.github = this.getGithubRepoInfo()
+        blackduckData.data.blackduck.automation.prcomment = true
+      }
     } else {
-      blackduckData.data.blackduck.automation.prcomment = false
-    }
-    if (!isPullRequestEvent() && parseToBoolean(inputs.BLACKDUCK_REPORTS_SARIF_CREATE)) {
-      const sarifReportFilterSeverities: string[] = []
-      if (inputs.BLACKDUCK_REPORTS_SARIF_SEVERITIES) {
-        const filterSeverities = inputs.BLACKDUCK_REPORTS_SARIF_SEVERITIES.split(',')
-        for (const fixPrSeverity of filterSeverities) {
-          if (fixPrSeverity != null && fixPrSeverity !== '') {
-            sarifReportFilterSeverities.push(fixPrSeverity.trim())
+      /** Set Black Duck Fix PR inputs in case of non PR context */
+      if (parseToBoolean(inputs.BLACKDUCK_FIXPR_ENABLED)) {
+        info('Black Duck Fix PR is enabled')
+        blackduckData.data.blackduck.fixpr = this.setBlackDuckFixPrInputs()
+        blackduckData.data.github = this.getGithubRepoInfo()
+      }
+      /** Set Black Duck SARIF inputs in case of non PR context */
+      if (parseToBoolean(inputs.BLACKDUCK_REPORTS_SARIF_CREATE)) {
+        const sarifReportFilterSeverities: string[] = []
+        if (inputs.BLACKDUCK_REPORTS_SARIF_SEVERITIES) {
+          const filterSeverities = inputs.BLACKDUCK_REPORTS_SARIF_SEVERITIES.split(',')
+          for (const sarifSeverity of filterSeverities) {
+            if (sarifSeverity != null && sarifSeverity !== '') {
```

Comment:
We can put condition like 'if(sarifSeverity)". This way condition checks for null, empty or undefined variable.

Reply:
updated